packages/expo-ui/src/jetpack-compose/layout.tsx

```diff
@@ -104,7 +105,11 @@ export type TextFontWeight =
   | '900';
 
 export type TextProps = {
-  children: string;
+  /**
+   * The children of the text.
+   * Only string and number are supported.
+   */
+  children?: React.ReactNode;
```

Comment:
Made it React.ReactNode. We can look into added Nested Text support at some point.